app/src/main/java/org/kiwix/kiwixmobile/nav/destination/library/LocalLibraryFragment.kt

```diff
@@ -18,38 +18,159 @@
+import org.kiwix.kiwixmobile.core.zim_manager.fileselect_view.adapter.BooksOnDiskAdapter
+import org.kiwix.kiwixmobile.core.zim_manager.fileselect_view.adapter.BooksOnDiskListItem
 import org.kiwix.kiwixmobile.local_file_transfer.LocalFileTransferActivity
+import org.kiwix.kiwixmobile.zim_manager.ZimManageViewModel
 import org.kiwix.kiwixmobile.zim_manager.ZimManageViewModel.FileSelectActions
 import org.kiwix.kiwixmobile.zim_manager.ZimManageViewModel.FileSelectActions.RequestMultiSelection
 import org.kiwix.kiwixmobile.zim_manager.ZimManageViewModel.FileSelectActions.RequestNavigateTo
 import org.kiwix.kiwixmobile.zim_manager.ZimManageViewModel.FileSelectActions.RequestSelect
 import org.kiwix.kiwixmobile.zim_manager.fileselect_view.FileSelectListState
-import org.kiwix.kiwixmobile.zim_manager.fileselect_view.ZimFileSelectFragment
+import javax.inject.Inject
 
-class LocalLibraryFragment : ZimFileSelectFragment() {
+private const val WAS_IN_ACTION_MODE = "WAS_IN_ACTION_MODE"
 
-  override val bookDelegate: BookOnDiskDelegate.BookDelegate by lazy {
+class LocalLibraryFragment : BaseFragment() {
+
+  @Inject lateinit var viewModelFactory: ViewModelProvider.Factory
+  @Inject lateinit var sharedPreferenceUtil: SharedPreferenceUtil
+
+  private var actionMode: ActionMode? = null
+  private val disposable = CompositeDisposable()
+
+  private val zimManageViewModel by lazy {
+    requireActivity().viewModel<ZimManageViewModel>(viewModelFactory)
+  }
+
+  private val bookDelegate: BookOnDiskDelegate.BookDelegate by lazy {
     BookOnDiskDelegate.BookDelegate(sharedPreferenceUtil,
       { offerAction(RequestNavigateTo(it)) },
       { offerAction(RequestMultiSelection(it)) },
       { offerAction(RequestSelect(it)) })
   }
+  private val booksOnDiskAdapter: BooksOnDiskAdapter by lazy {
+    BooksOnDiskAdapter(bookDelegate, BookOnDiskDelegate.LanguageDelegate)
+  }
+
+  override fun onDestroyView() {
+    super.onDestroyView()
+    actionMode = null
+    disposable.clear()
+  }
+
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+    zim_swiperefresh.setOnRefreshListener(::requestFileSystemCheck)
+    zimfilelist.run {
+      adapter = booksOnDiskAdapter
+      layoutManager = LinearLayoutManager(context, RecyclerView.VERTICAL, false)
+      setHasFixedSize(true)
+    }
+    zimManageViewModel.fileSelectListStates.observe(viewLifecycleOwner, Observer(::render))
+    disposable.add(sideEffects())
+    zimManageViewModel.deviceListIsRefreshing.observe(viewLifecycleOwner, Observer {
+      zim_swiperefresh.isRefreshing = it!!
+    })
+    if (savedInstanceState != null && savedInstanceState.getBoolean(WAS_IN_ACTION_MODE)) {
+      zimManageViewModel.fileSelectActions.offer(FileSelectActions.RestartActionMode)
+    }
+
+    disposable.add(zimManageViewModel.libraryTabIsVisible.subscribe { finishActionMode() })
```

Comment:
this strikes me as useless now, was only useful when the tabs existed

Reply:
It sure does..